### PR TITLE
Support `PREWHERE` condition for ClickHouse dialect

### PR DESCRIPTION
**Pull request by git-hulk**
For ClickHouse PREWHERE clause, please refer to:

https://clickhouse.com/docs/en/sql-reference/statements/select/prewhere

### CODE REVIEW

File: tests/sqlparser_clickhouse.rs [hunk 1]
```diff
@@ -681,6 +682,56 @@ fn parse_group_by_with_modifier() {
     }
 }
 
+#[test]
+fn test_prewhere() {
+    match clickhouse().verified_stmt("SELECT * FROM t PREWHERE x = 1 WHERE y = 2") {
```
Comment (iffyio) [inline tests/sqlparser_clickhouse.rs:687]:
```suggestion
    match clickhouse_and_generic().verified_stmt("SELECT * FROM t PREWHERE x = 1 WHERE y = 2") {
```

File: tests/sqlparser_clickhouse.rs [hunk 2]
```diff
@@ -681,6 +682,56 @@ fn parse_group_by_with_modifier() {
...
+                    left: Box::new(Identifier(Ident::new("y"))),
+                    op: BinaryOperator::Eq,
+                    right: Box::new(Expr::Value(Value::Number("2".parse().unwrap(), false))),
+                })
+            );
+        }
+        _ => unreachable!(),
+    }
+
+    match clickhouse().verified_stmt("SELECT * FROM t PREWHERE x = 1 AND y = 2") {
```
Comment (iffyio) [inline tests/sqlparser_clickhouse.rs:711]:
```suggestion
    match clickhouse_and_generic().verified_stmt("SELECT * FROM t PREWHERE x = 1 AND y = 2") {
```

**Review by iffyio — APPROVED** [review]
left a couple minor edits to cover generic in the test case, otherwise LGTM!
cc @alamb

**Comment by git-hulk** [comment]
@iffyio Fixed, thanks!

**Review by alamb — APPROVED** [review]
Thank you @git-hulk and @iffyio

**Pull request merged**